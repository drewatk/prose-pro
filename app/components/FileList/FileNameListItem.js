import React from "react";
import {
  ListGroupItem,
  Dropdown,
  DropdownToggle,
  DropdownItem,
  DropdownMenu
} from "reactstrap";

class FileNameListItem extends React.Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      dropdownOpen: false
    };
  }

  toggle() {
    this.setState(prevState => ({
      dropdownOpen: !prevState.dropdownOpen
    }));
  }

  render() {
    const { file, onFileItemClick, styles } = this.props;
    return (
      <ListGroupItem
        id="file-item"
        onClick={() => onFileItemClick(file)}
        style={styles}
      >
        {file}
        <Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
          <DropdownToggle tag="span">
            <i className="fas fa-ellipsis-h" style={{ float: "right" }} />
          </DropdownToggle>
          <DropdownMenu>
            <DropdownItem>Delete File</DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </ListGroupItem>
    );
  }
}

export default FileNameListItem;
