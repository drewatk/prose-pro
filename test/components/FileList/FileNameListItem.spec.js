import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import FileNameListItem from "app/components/FileList/FileNameListItem";

Enzyme.configure({ adapter: new Adapter() });

describe("FileNameListItem", () => {
  it("matches snapshot", () => {
    const props = {
      file: "file1",
      onFileItemClick: jest.fn(),
      onFileDeleteClick: jest.fn()
    };
    const wrapper = shallow(<FileNameListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("shows dropdown on click, and calls callback", () => {
    const props = {
      file: "file1",
      onFileItemClick: jest.fn(),
      onFileDeleteClick: jest.fn()
    };
    const wrapper = mount(<FileNameListItem {...props} />);
    wrapper.find("i").simulate("click");

    expect(wrapper.find("DropdownItem")).toHaveLength(1);

    wrapper.find("DropdownItem").simulate("click");
    expect(props.onFileItemClick).toBeCalledTimes(0);
    expect(props.onFileDeleteClick).toBeCalledTimes(1);
  });

  it("calls callback", () => {
    const props = {
      file: "file1",
      onFileItemClick: jest.fn(),
      onFileDeleteClick: jest.fn()
    };
    const wrapper = mount(<FileNameListItem {...props} />);
    wrapper.find("i").simulate("click");

    expect(wrapper.find("ListGroupItem")).toHaveLength(1);

    wrapper.find("ListGroupItem").simulate("click");

    expect(props.onFileItemClick).toBeCalledTimes(1);
    expect(props.onFileDeleteClick).toBeCalledTimes(0);
  });
});
