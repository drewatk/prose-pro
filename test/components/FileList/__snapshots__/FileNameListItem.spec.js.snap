// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileNameList matches snapshot 1`] = `
<ListGroupItem
  id="file-item"
  onClick={[Function]}
  tag="li"
>
  file1
  <Dropdown
    active={false}
    addonType={false}
    direction="down"
    inNavbar={false}
    isOpen={false}
    nav={false}
    setActiveFromChild={false}
    toggle={[Function]}
  >
    <DropdownToggle
      aria-haspopup={true}
      color="secondary"
      onClick={[Function]}
      tag="span"
    >
      <i
        className="fas fa-ellipsis-h"
        style={
          Object {
            "float": "right",
          }
        }
      />
    </DropdownToggle>
    <DropdownMenu
      flip={true}
      tag="div"
    >
      <DropdownItem
        onClick={[Function]}
        tag="button"
        toggle={true}
      >
        Delete File
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
</ListGroupItem>
`;

exports[`FileNameListItem matches snapshot 1`] = `
<ListGroupItem
  id="file-item"
  onClick={[Function]}
  tag="li"
>
  file1
  <Dropdown
    active={false}
    addonType={false}
    direction="down"
    inNavbar={false}
    isOpen={false}
    nav={false}
    setActiveFromChild={false}
    toggle={[Function]}
  >
    <DropdownToggle
      aria-haspopup={true}
      color="secondary"
      onClick={[Function]}
      tag="span"
    >
      <i
        className="fas fa-ellipsis-h"
        style={
          Object {
            "float": "right",
          }
        }
      />
    </DropdownToggle>
    <DropdownMenu
      flip={true}
      tag="div"
    >
      <DropdownItem
        onClick={[Function]}
        tag="button"
        toggle={true}
      >
        Delete File
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
</ListGroupItem>
`;
